src/contrib/fbgemm/fully_connected_int8.cc

```diff
@@ -0,0 +1,106 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file Use external fbgemm library call.
+ */
+
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/util.h>
+#include <dmlc/logging.h>
+#include <fbgemm/Fbgemm.h>
+#include <fbgemm/FbgemmFP16.h>
+#include <fbgemm/QuantUtilsAvx2.h>
+#include <random>
+#include "fbgemm_utils.h"
+#include <cpuinfo.h>
+
+namespace tvm {
+namespace contrib {
+
+using namespace runtime;
+using namespace fbgemm;
+
+
+TVM_REGISTER_GLOBAL("tvm.contrib.fbgemm.fully_connected_int8")
+.set_body([](TVMArgs args, TVMRetValue *ret) {
```

Comment:
There are some computation that needs to be factored out as a pre-processing step like ComputeColumnOffsets and PackBMatrix . Otherwise, those pre-processing computation overhead will dominate the total time negating speedups from fbgemm.

Reply:
Yes, indeed. I just measure the time of fbgemm_packed/packA/Requant... I probably need to divide it into three external calls. So many arguments: :(.